metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESBrowseDAO.java

```diff
@@ -336,4 +360,143 @@ public List<String> getBrowsePaths(@Nonnull String entityName, @Nonnull Urn urn)
+    try {
+      final SearchResponse groupsResponse;
+      try (Timer.Context ignored = MetricUtils.timer(this.getClass(), "esGroupSearch").time()) {
+        final String finalInput = input.isEmpty() ? "*" : input;
+        groupsResponse =
+            client.search(constructGroupsSearchRequestV2(entityName, path, filter, finalInput), RequestOptions.DEFAULT);
+      }
+
+      final BrowseGroupsResultV2 browseGroupsResult = extractGroupsResponseV2(groupsResponse, path, start, count);
+      final int numGroups = browseGroupsResult.getTotalGroups();
+
+      return new BrowseResultV2().setMetadata(
+              new BrowseResultMetadata().setTotalNumEntities(browseGroupsResult.getTotalNumEntities()).setPath(path))
+          .setGroups(new BrowseResultGroupV2Array(browseGroupsResult.getGroups()))
+          .setNumGroups(numGroups)
+          .setFrom(start)
+          .setPageSize(count);
+    } catch (Exception e) {
+      log.error("Browse V2 query failed: " + e.getMessage());
+      throw new ESQueryException("Browse V2 query failed: ", e);
+    }
+  }
+
+  @Nonnull
+  private SearchRequest constructGroupsSearchRequestV2(@Nonnull String entityName, @Nonnull String path, @Nullable Filter filter, @Nonnull String input) {
+    final String indexName = indexConvention.getIndexName(entityRegistry.getEntitySpec(entityName));
+    final SearchRequest searchRequest = new SearchRequest(indexName);
+    final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+    searchSourceBuilder.size(0);
+    searchSourceBuilder.query(buildQueryStringV2(entityName, path, SearchUtil.transformFilterForEntities(filter, indexConvention), input));
+    searchSourceBuilder.aggregation(buildAggregationsV2(path));
+    searchRequest.source(searchSourceBuilder);
+    return searchRequest;
+  }
+
+  /**
+   * Extracts the name of group from path.
+   *
+   * <p>Example: ␟foo␟bar␟baz => baz
+   *
+   * @param path path of the group/entity
+   * @return String
+   */
+  @Nonnull
+  private String getSimpleNameV2(@Nonnull String path) {
+    return path.substring(path.lastIndexOf(BROWSE_V2_DELIMITER) + 1);
+  }
+
+
+  private static int getPathDepthV2(@Nonnull String path) {
+    return StringUtils.countMatches(path, BROWSE_V2_DELIMITER);
+  }
+
+  @Nonnull
+  private QueryBuilder buildQueryStringV2(@Nonnull String entityName, @Nonnull String path, @Nullable Filter filter, @Nonnull String input) {
+    final int browseDepthVal = getPathDepthV2(path);
+
+    final BoolQueryBuilder queryBuilder = QueryBuilders.boolQuery();
+
+    EntitySpec entitySpec = entityRegistry.getEntitySpec(entityName);
+    QueryBuilder query = SearchRequestHandler
+        .getBuilder(entitySpec, searchConfiguration, customSearchConfiguration)
+        .getQuery(input, false);
+    queryBuilder.must(query);
+
+    queryBuilder.mustNot(QueryBuilders.termQuery(REMOVED, "true"));
```

Comment:
advanced filters lets users filter for "removed" (soft deleted) entities. im wondering how this will impact what we are seeing here..

Reply:
that's a fantastic point. I think we should do what we do for regular search, add this as a default unless they're passing in `REMOVED` in the list of filters which they could do in advanced. will make that change!